### PR TITLE
Backport Wallet refactoring and external wallet files 11687 and 13017

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
~Currently based on https://github.com/dashpay/dash/pull/3412~

~Will be rebased once that is merged.~

This is a dependency for backporting #10740

In the mean time, I need some advise from @UdjinM6 as to how I should fix the autobackup system now that the wallet files are "named" "" if they aren't specified by `-wallet` 
I tried moving the InitAutoBackup call in init.cpp down below wallets are opened, and then running backups via CWallet* and not by name. However, when I did this it resulted in no backups being created. (I also tried a bunch of other methods, but each had their own problems associated)

Thanks

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
This breaks auto-backup functionality, need some advise regarding that

File: doc/release-notes.md [hunk 1]
```diff
@@ -465,6 +465,36 @@ See detailed [set of changes](https://github.com/dashpay/dash/compare/v0.14.0.5.
...
+For backwards compatibility, wallet paths that are names of existing data files
+in the `-walletdir` directory will continue to be accepted and interpreted the
+same as before.
+
+Low-level RPC changes
+---------------------
+
+- When Dash Core is not started with any `-wallet=<path>` options, the name of
+  the default wallet returned by `getwalletinfo` and `listwallets` RPCs is
+  now the empty string `""` instead of `"wallet.dat"`. If bitcoin is started
```
Comment (thephez) [inline doc/release-notes.md:494]: Bitcoin reference

**Comment by PastaPastaPasta** [comment]
@UdjinM6 starting dashcore `./src/qt/dash-qt --testnet` which opens wallet.dat results in a backup being created, however, `./src/qt/dash-qt --testnet --wallet=wallet.dat` does not result in a backup

File: doc/release-notes.md (hunk 1 above)
Comment (PastaPastaPasta) [reply doc/release-notes.md]: https://github.com/dashpay/dash/compare/87068fe5b82f8276af7904ed3257cf8a397631bc..97907b051ffe7097e2d86bb7ce9e945f4d95de3f

**Comment by UdjinM6** [comment]
Good catch! 3a20581b94 should fix that.

**Comment by PastaPastaPasta** [comment]
Cherry-picked

**Pull request merged**